app/queries/work_unit_query.rb

```diff
@@ -0,0 +1,50 @@
+class WorkUnitQuery
+
+  def initialize(user, start_date, end_date, kind)
```

Comment:
I'd like you to work on making this method less complex.  I think you can actually get it down to about 3 lines, and those lines being simpler :-) 

Some tidbits/suggestions:
1) you've effectively written the function twice ... once for String arguments and once for Date arguments.   Try supporting just one argument type (I would recommend Date instead of String), that will cut the size of the function in half.

2) A pattern you've used frequently throughout the code (not just here) is to convert a String object to a DateTime object with #parse, and then convert back to a string with #strftime.  Ruby's DateTime, Date, and Time classes are remarkably versatile: you can do everything you want to do without ever converting them to Strings until the last moment when you output them.   It's worth taking a few minutes to read over some of the documentation for DateTime, Date, and Time, to see what methods are available (noting that DateTime inherits most of the methods of both Date and  Time):

* http://ruby-doc.org/core-2.1.2/Time.html
* http://ruby-doc.org/stdlib-2.1.2/libdoc/date/rdoc/Date.html
* http://ruby-doc.org/stdlib-2.1.2/libdoc/date/rdoc/DateTime.html
       
3) A specific example: the method #cdway (which is available for both Date and DateTime) allows you to check which day of the week a date is (Stands for 'calendar week day').  #cwday returns 1 for Monday and 7 for Sunday, so <some date>.cwday == 7 has the same result thing as <some date>.strftime('%A') == 'Sunday' without performing the String conversion.   You could set a constant SUNDAY == 7   if you wanted to make the code say <date>.cwday == SUNDAY, which is a bit more readable than the bare number 7.      

4) Another example: in hours_reports_controller.rb on line 20 you iterate a group of DateTime objects and convert each to a String with strftime to store in the array of Sundays.  Then later, those strings get pulled back out of the array and passed to WorkUnitQuery ... where they get parsed back into dates.  Try just storing the DateTime objects in the array instead.

5) You don't need "elsif" if the else condition is the exact inverse of the "if" condition.  i.e.:

    if thing_is_true
      ... do something ...
    elsif !thing_is_true
      ... something else ...
    end

Has exactly the same behavior as:

    if thing_is_true
      ... do something ...
    else
      ... something else ...
    end

But is a bit easier to read and executes a little faster (because the comparison doesn't need to be made a second time).